src/nodes/accessors/Lights.js

```diff
@@ -32,8 +32,18 @@ export function lightShadowMatrix( light ) {
 
 	return data.shadowMatrix || ( data.shadowMatrix = uniform( 'mat4' ).setGroup( renderGroup ).onRenderUpdate( ( frame ) => {
 
+		// normally, shadow matrices are updated in ShadowNode. However, if the shadow matrix is used outside
+		// of shadow rendering (like in ProjectorLightNode), the shadow matrix still requires an update
+
 		if ( light.castShadow !== true || frame.renderer.shadowMap.enabled === false ) {
 
+			if ( light.shadow.camera.coordinateSystem !== frame.camera.coordinateSystem ) {
```

Comment:
The problem is we can't fix the projection matrix in `LightShadow.updateMatrices()` because there is no reference to the coordinate system. So all callers of `LightShadow.updateMatrices()` must ensure the shadow camera's projection matrix is correct.